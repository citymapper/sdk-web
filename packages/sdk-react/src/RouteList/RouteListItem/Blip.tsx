import React from 'react'
import IconBlip from '../../Icons/IconBlip'
import styled from '../../utils/cmStyled'

const BlipStyle = styled('div', { className: 'cm-departure-live-blip' })`
  display: flex;
  align-items: flex-start;
  align-self: flex-start;
  margin-right: ${({ theme }) => theme.spacing(0.5)};
  svg {
    width: 8px;
    height: 8px;
  }
  &.cm-departure-live-blip {
    ${({ theme }) => theme.CmDeparturesLive?.blip}
  }
`

const Blip = () => (
  <BlipStyle>
    <IconBlip />
  </BlipStyle>
)

export default Blip
