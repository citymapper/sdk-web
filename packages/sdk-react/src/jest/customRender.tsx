import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
