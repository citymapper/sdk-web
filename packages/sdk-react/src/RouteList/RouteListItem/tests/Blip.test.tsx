import React from 'react'
import customRender from '../../../jest/customRender'
import Blip from '../Blip'

describe('Blip', () => {
  it('should match snapshot', () => {
    const { container } = customRender(<Blip />)
    expect(container).toMatchSnapshot()
  })
})
