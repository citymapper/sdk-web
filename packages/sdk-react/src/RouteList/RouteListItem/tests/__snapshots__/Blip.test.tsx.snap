// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blip should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-right: 4px;
}

.emotion-0 svg {
  width: 8px;
  height: 8px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
}

<div>
  <div
    class="cm-departure-live-blip emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      viewBox="0 0 8 8"
    >
      <path
        clip-rule="evenodd"
        d="M3.46447 3.46447C4.40215 2.52678 5.67392 2 7 2C7.55228 2 8 1.55228 8 1C8 0.447715 7.55228 0 7 0C5.14348 0 3.36301 0.737498 2.05025 2.05025C0.737498 3.36301 0 5.14348 0 7C0 7.55228 0.447715 8 1 8C1.55228 8 2 7.55228 2 7C2 5.67392 2.52678 4.40215 3.46447 3.46447ZM5.93934 5.93934C6.22064 5.65804 6.60218 5.5 7 5.5C7.55228 5.5 8 5.05228 8 4.5C8 3.94772 7.55228 3.5 7 3.5C6.07174 3.5 5.1815 3.86875 4.52513 4.52513C3.86875 5.1815 3.5 6.07174 3.5 7C3.5 7.55228 3.94772 8 4.5 8C5.05228 8 5.5 7.55228 5.5 7C5.5 6.60218 5.65804 6.22064 5.93934 5.93934Z"
        fill="#FFD600"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
